### ISSUE TITLE
floatingFilterComponent does not accept string like in example

### ISSUE THREAD

**Issue by Ketec** [ag-grid/ag-grid#2193]
**I'm submitting a ...**  (check one with "x")
```
[ x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/ceolter/ag-grid/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
floatingFilterComponent?: {new(): IFloatingFilterComp<any, any, any>};

Expects a component that implements this interface. Passing a string like: 
floatingFilterComponent: "sliderFloatingFilter" is not valid

` Type 'string' is not comparable to type 'new () => IFloatingFilterComp<any, any, any>'.`

**Expected behavior**
Accepts string if it is already added under frameworkComponents.

**Comment by VictorNovice** [comment]
It actually works ok if you update the definition file to accept string.
I think there's a mistake in the definition file.

**Comment by makinggoodsoftware** [comment]
Hi,

In ag-grid we have recently started processing github issues as well as customer issues. We are also going through the backlog of issues already open and acting on them as necessary, on this particular case, it seems like this issue could be gone in our latest release.

Could you update the ag-grid version in your end and try again?

Note that this issue is getting closed, but you are welcomed to create a new issue if you think this issue is still not resolved

Thanks

**Comment by JayceeZ** [comment]
Hello, I checked and it hasn't been fixed, even in 19.1.1

Can you target a fix of that in 17.1.x onwards ?

Thank you.

**Comment by JayceeZ** [comment]
Let me open a new ticket.
